observ/proxy.py

```diff
@@ -33,7 +36,7 @@ def __del__(self):
 TYPE_LOOKUP = {}
 
 
-def proxy(target, readonly=False, shallow=False):
+def proxy(target: T, readonly=False, shallow=False) -> T:
```

Comment:
Shouldn't this be `-> Proxy[T]`?

Reply:
> Shouldn't this be -> Proxy[T]

For a `Proxy[T]` to be useful as a return type, you would need a way to express the fact that a `Proxy[T]` delegates all methods to its inner `T`, which is not possible in the current state of the Python type system AFAICT.

Reply:
I see, that's a shame. Either way, observ provides fully transparent wrappers over the builtin types, so it should be fine! :)